util/net.go

```diff
@@ -0,0 +1,35 @@
+package netutils
```

Comment:
I agree with your dislike regarding `util`, `common` etc. It might be nice to have the same logic for extracting the source of a request in multiple places. But I'm fine with inlining this for now inside the source predicate